### PR TITLE
Feature/search sort user roles TM-2423

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
Dual Merge: [BE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/644)

~There's a css issue - can u find it?~ 🦜 

'al' is a good search term for testing the first name || last name search

### CODE REVIEW

File: src/Components/AdministratorPage/UserRoles/UserRoles.jsx [hunk 1]
```diff
@@ -54,75 +101,145 @@ class UserRoles extends Component {
...
+
+      return (
+        <div className="delegate-role-header-sort">
+          {
+            <FA name={`long-arrow-${direction}`} />
+          }
+        </div>
+      );
+    };
+
```
Comment (scott062) [inline src/Components/AdministratorPage/UserRoles/UserRoles.jsx:136]: You could simplify and store the sort and icon in a single object with a single return statement calling the [key] lookup

File: src/actions/userRoles.js [hunk 2]
```diff
@@ -23,8 +23,11 @@ export function usersSuccess(results) {
   };
 }
 
-export function getUsers(page = 1, limit = 100) {
-  const qString = queryString.stringify({ page, limit });
+export function getUsers(page = 1, limit = 100, sort, filters, q_username, q_name) {
+  const params = { page, limit, sort, filters, q_username, q_name };
+  // eslint-disable-next-line no-unused-vars
```
Comment (scott062) [inline src/actions/userRoles.js:28]: Needed?

Comment (elizabeth-jimenez) [reply src/actions/userRoles.js:28]: yea 😭

File: src/Components/AdministratorPage/UserRoles/UserRoles.jsx [hunk 3]
```diff
@@ -17,115 +22,207 @@ class UserRoles extends Component {
...
-      return omit(roles, removeRoles);
-    };
+  filterByPermission = (clicked, permission) => {
+    const stateFilters = this.state.filters;
+    if (clicked) {
+      stateFilters.push(permission);
+    } else {
+      pull(stateFilters, permission);
+    }
+    this.setState({ filters: stateFilters }, this.callUpdateUsers);
```
Comment (mjoyce91) [inline src/Components/AdministratorPage/UserRoles/UserRoles.jsx:51]: you can do `const { filters } = this.state`

Comment (elizabeth-jimenez) [reply src/Components/AdministratorPage/UserRoles/UserRoles.jsx]: i really wanna call it stateFilters, for some reason lol - lemme know if the compromise is alright

**Review by scott062 — APPROVED** [review]

**Pull request merged**